src/main/scala/nl/knaw/dans/easy/ingest/EasyIngest.scala

```diff
@@ -208,12 +215,19 @@ object EasyIngest {
       System.exit(13)
     }
 
+  private class CompositeException(throwables: List[Throwable]) extends RuntimeException(throwables.foldLeft("")((msg, t) => s"$msg\n${getMessage(t)} ${getStackTrace(t)}"))
```

Comment:
That's right, but it is exposed through `run` inside a `Try`, right? If you use `recover` or `recoverWith` on the result of calling `run`, you **NEED** to do pattern matching since the argument of these operators is a **`PartialFunction`**! I'm not sure if you can do this when the expected exception is declared private.

```scala
XXX.run()
   .recoverWith {
       case e: CompositeException => ??? // can you do this with a private class? I don't actually know, but I expect not...
       case CompositeException(errors) => ??? // this one only works if you would declare it to be a case class, which is currently not the case
   }
```